text/tls-and-alpn.rst

```diff
@@ -0,0 +1,380 @@
+| <default>         | No ALPN protocol match     | ``/server/status/ready``           |                                    |
+|                   |                            |                                    |                                    |
+|                   |                            | ``/db/<database>/edgeql?query=``   |                                    |
+|                   |                            |                                    |                                    |
+|                   |                            | ``/db/<database>/graphql/explore`` |                                    |
+|                   |                            |                                    |                                    |
+|                   |                            | ``/db/<database>/notebook/status`` |                                    |
++-------------------+----------------------------+------------------------------------+------------------------------------+
+
+On the client side for either Option, the bindings or the CLI should provide an
+ALPN protocol based on the scenario. In most cases, the client binding should
+use ``edgedb-binary``
+
+
+Client Bindings & REPL
+----------------------
+
+The official client bindings for Python, Go and Node.js, as well as the EdgeDB
+CLI (the REPL and any command connecting to the server) will use TLS to talk to
+the EdgeDB instances by default. They should advertise the expected ALPN
+protocol in the TLS handshake, and be ready to handle any negotiation result,
+depending on the conclusion we'll make for previous questions.
+
+For compatibility considerations, the client is recommended to provide a
+temporary option to connect to old EdgeDB servers without TLS. However, this
+option should come with a warning and should not be default.
+
+For certificate verification, this was discussed previously. To recap, in order
+to make the client code simple and consistent, the client should either check
+the well-known EdgeDB location for the flag of self-signed certificate in order
+to trust the certificate, or discover the local CA and trust the CA certificate.
+The alternative is to enforce the user to explicitly turn on a switch to allow
+self-signed certificates, but this depends on the result of the discussion.
+
+
+Compatibility & Development
+---------------------------
+
+We mentioned that the new client is recommended to have an option (with
+warnings) to connect to old EdgeDB servers without TLS. In the other hand, we
+won't allow old EdgeDB clients connecting to newer EdgeDB servers with TLS
+enabled. On a TLS handshake failure, the EdgeDB server will return a error in
+clear text, indicating that the client must be upgraded to use TLS.
+
+However, the core EdgeDB server (``edb server``) will provide a hidden option
+``--no-tls`` to run the server in non-TLS compatible mode for development and
```

Comment:
or perhaps we just use unix sockets for local dev...

Reply:
Let's just make `edb server` initialize self-signed certificate and put credentials file like `local-dev.json` (or whatever). So you can use `edgedb -Ilocal-dev` to connect to `edb server`.